test/Mono.Linker.Tests.Cases/Warnings/WarningSuppression/DetectRedundantSuppressionsFeatureSubstitutions.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq.Expressions;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.Warnings.WarningSuppression
+{
+	[SetupLinkerSubstitutionFile ("DetectRedundantSuppressionsFeatureSubstitutions.xml")]
+	[SetupLinkerArgument ("--feature", "Feature", "false")]
+	[ExpectedNoWarnings]
+	[SkipKeptItemsValidation]
+	public class DetectRedundantSuppressionsFeatureSubstitutions
+	{
+		// https://github.com/dotnet/linker/issues/2920
+		public static void Main ()
+		{
+			ReportRedundantSuppressionWhenTrimmerIncompatibleCodeDisabled.Test ();
+			DoNotReportUsefulSuppressionWhenTrimmerIncompatibleCodeEnabled.Test ();
+		}
+
+		public static Type TriggerUnrecognizedPattern ()
+		{
+			return typeof (DetectRedundantSuppressionsFeatureSubstitutions);
+		}
+
+		public static string TrimmerCompatibleMethod ()
+		{
+			return "test";
+		}
+
+		public static bool IsFeatureEnabled {
+			get => throw new NotImplementedException ();
+		}
+
+		class ReportRedundantSuppressionWhenTrimmerIncompatibleCodeDisabled
+		{
+			[ExpectedWarning ("IL2121", "IL2072")]
```

Comment:
I would add a comment here referencing the issue and a short description of why it's currently wrong (imagine coming to this code next year and trying to figure out what the test was about)

Reply:
Added comment and moved the link here.